src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -88,10 +88,11 @@ def _migrate_pipelines(self) -> list[partial[dict | bool | list | BinaryIO]]:
 
         tasks = []
         for pipeline in pipelines_to_migrate:
-            if pipeline.pipeline_id in self._skip_pipeline_ids:
-                logger.info(f"Skipping pipeline {pipeline.pipeline_id}")
-                continue
-            tasks.append(partial(self._migrate_pipeline, pipeline))
+            if len(self._include_pipeline_ids) > 0:
+                if pipeline.pipeline_id in self._include_pipeline_ids:
+                    tasks.append(partial(self._migrate_pipeline, pipeline))
```

Comment:
We like to be able to tell a difference between "include pipelines is not specified" and "there are no pipeline to include". The former means that we include **all** pipelines, the later that we include **no** pipelines. This [allows us to skip the pipeline completely](https://github.com/databrickslabs/ucx/pull/3489), which might be useful for some users, but is definitely useful during migration tests.

```suggestion
            if self._include_pipeline_ids is not None and pipeline.pipeline_id in self._include_pipeline_ids:
                tasks.append(partial(self._migrate_pipeline, pipeline))
```